Fails to build on x86 without SSE2

FreeBSD, NetBSD, OpenBSD still support pre-SSE2 CPUs, so downstream `rust` package for 32-bit x86 (aka i386) targets real i686 (aka `pentiumpro`) or i586 (aka `pentium`). To reproduce on Linux (via rustup) pass `--target i586-unknown-linux-gnu` or similar.

```rust
$ cargo build
[...]
error[E0425]: cannot find function `vectorization_support_no_cache_x86` in this scope
  --> src/lib.rs:63:28
   |
63 |         let val = unsafe { vectorization_support_no_cache_x86() };
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

For more information about this error, try `rustc --explain E0425`.
error: could not compile `faster-hex` (lib) due to previous error
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -71,10 +71,7 @@ pub(crate) fn vectorization_support() -> Vectorization {
 
 // We enable xsave so it can inline the _xgetbv call.
 #[target_feature(enable = "xsave")]
-#[cfg(all(
-    any(target_arch = "x86", target_arch = "x86_64"),
-    target_feature = "sse"
-))]
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
 #[cold]
 unsafe fn vectorization_support_no_cache_x86() -> Vectorization {
     #[cfg(target_arch = "x86")]
@@ -127,6 +124,7 @@ mod tests {
                 Vectorization::AVX2 => assert!(is_x86_feature_detected!("avx2")),
                 Vectorization::SSE41 => assert!(is_x86_feature_detected!("sse4.1")),
                 Vectorization::None => assert!(
+                    !cfg!(target_feature = "sse") ||
                     !is_x86_feature_detected!("avx2") && !is_x86_feature_detected!("sse4.1")
                 ),
             }

